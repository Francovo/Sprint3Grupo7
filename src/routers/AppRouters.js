import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import NavBar from "../components/NavBar/NavBar";
import { useEffect, useState } from "react";
import { getAuth, onAuthStateChanged } from "firebase/auth";
import RouterPublic from "./RouterPublic";
import { PrivateRoute } from "./PrivateRoute";
import RouterPrivate from "./RouterPrivate";
import { PublicRoute } from "./PublicRouter";

function AppRouters() {
  const [logged, setLogged] = useState(false);
  const [checking, setChecking] = useState(true);

  useEffect(() => {
    const auth = getAuth();
    onAuthStateChanged(auth, (user) => {
      if (user?.uid) {
        console.log(user);
        console.log("Logueado");
        setLogged(true);
      } else {
        console.log("no log");
        setLogged(false);
      }
      setChecking(false);
    });
  }, [setLogged, setChecking]);
  if (checking) {
    return <h1>Espere...</h1>;
  }

  return (
    <div>
      <NavBar />
      <Router>
        <Routes>
          <Route
            path="*"
            isAuthenticated={logged}
            element={
              <PublicRoute>
                <RouterPublic />
              </PublicRoute>
            }
          />

          <Route
            path="/favoritos"
            isAuthenticated={logged}
            element={
              <PrivateRoute>
                <RouterPrivate />
              </PrivateRoute>
            }
          />
        </Routes>
      </Router>
    </div>
  );
}

export default AppRouters;
