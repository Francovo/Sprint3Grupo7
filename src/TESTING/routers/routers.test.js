import { shallow } from "enzyme";
import AppRouters from "../../routers/AppRouters";

describe("router publico y privado", () => {
  test("isLogged", () => {
    const wrapper = shallow(<AppRouters />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("PublicRoute")).toContain(false);
  });
});
