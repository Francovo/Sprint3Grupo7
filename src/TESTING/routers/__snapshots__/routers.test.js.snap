// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router publico y privado isLogged 1`] = `
<div>
  <BrowserRouter>
    <Routes>
      <Route
        element={
          <PublicRoute
            isAuthenticated={false}
          >
            <NavBarPublic
              setSearch={[Function]}
            />
            <Registro />
          </PublicRoute>
        }
        path="/registro"
      />
      <Route
        element={
          <PublicRoute
            isAuthenticated={false}
          >
            <NavBarPublic
              setSearch={[Function]}
            />
            <Login />
          </PublicRoute>
        }
        path="/login"
      />
      <Route
        element={
          <PublicRoute
            isAuthenticated={false}
          >
            <NavBarPublic
              setSearch={[Function]}
            />
            <Home
              search={Array []}
            />
          </PublicRoute>
        }
        path="/home"
      />
      <Route
        element={
          <PublicRoute
            isAuthenticated={false}
          >
            <NavBarPublic
              setSearch={[Function]}
            />
            <MasValoradas />
          </PublicRoute>
        }
        path="/valoradas"
      />
      <Route
        element={
          <PrivateRoute
            isAuthenticated={false}
          >
            <RouterPrivate
              search={Array []}
              setSearch={[Function]}
            />
          </PrivateRoute>
        }
        path="/*"
      />
    </Routes>
  </BrowserRouter>
</div>
`;
